### ISSUE TITLE
[2.x] Revisions won't be created

### ISSUE THREAD

**Issue by ErikMichelson** [hedgedoc/hedgedoc#4262]
### Description

When editing a note and navigating back to the intro page, no revision will be saved which results in data loss.

### Steps to reproduce

1. Open a new note
2. Write some characters
3. Set an alias
4. Write some characters again
5. Click the HedgeDoc logo in the top left to navigate to the intro page
6. Visit the note again, all typed characters are lost. No revision was created.

### Expected behaviour

When navigating away from the editor, a new revision should be created and saved.

### Logs

```
@hedgedoc/backend:start:dev: 06/20/2023, 9:15:06 AM [HedgeDoc.RealtimeNote 1()] New realtime session for note 1 created. Length of initial content: 0 characters
@hedgedoc/backend:start:dev: 06/20/2023, 9:15:06 AM [HedgeDoc.RealtimeNote 1()] User 'Erik' connected
@hedgedoc/backend:start:dev: 06/20/2023, 9:15:06 AM [TypeORM.query()] SELECT "User"."id" AS "User_id", "User"."username" AS "User_username", "User"."displayName" AS "User_displayName", "User"."createdAt" AS "User_createdAt", "User"."updatedAt" AS "User_updatedAt", "User"."photo" AS "User_photo", "User"."email" AS "User_email" FROM "user" "User" WHERE ("User"."username" = ?) LIMIT 1 -- PARAMETERS: ["erik"]
@hedgedoc/backend:start:dev: 06/20/2023, 9:15:06 AM [TypeORM.query()] UPDATE "session" SET "expiredAt" = 1688462106790 WHERE "id" IN (?) -- PARAMETERS: ["yMovjzqJqY2LJ1lTFvYIaEaNVo5UztUa"]
@hedgedoc/frontend:start:dev: [2023-06-20 09:15:07] (Base URL Configuration) HD_RENDERER_BASE_URL is unset. Using editor base url for renderer.
@hedgedoc/frontend:start:dev: [2023-06-20 09:15:07] (Base URL Configuration) Editor base URL http://localhost:8080/
@hedgedoc/frontend:start:dev: [2023-06-20 09:15:07] (Base URL Configuration) Renderer base URL http://localhost:8080/
@hedgedoc/frontend:start:dev: [2023-06-20 09:15:07] (Frontend config fetcher) Fetched frontend config from http://localhost:8080/
@hedgedoc/backend:start:dev: 06/20/2023, 9:15:08 AM [HedgeDoc.RealtimeNote 1()] User 'Erik' disconnected. 0 clients left.
@hedgedoc/backend:start:dev: 06/20/2023, 9:15:08 AM [HedgeDoc.RealtimeNote 1()] Destroying realtime note.
@hedgedoc/backend:start:dev: 06/20/2023, 9:15:08 AM [TypeORM.query()] SELECT DISTINCT "distinctAlias"."Revision_id" AS "ids_Revision_id", "distinctAlias"."Revision_createdAt", "distinctAlias"."Revision_id" FROM (SELECT "Revision"."id" AS "Revision_id", "Revision"."patch" AS "Revision_patch", "Revision"."content" AS "Revision_content", "Revision"."length" AS "Revision_length", "Revision"."yjsStateVector" AS "Revision_yjsStateVector", "Revision"."createdAt" AS "Revision_createdAt", "Revision"."noteId" AS "Revision_noteId" FROM "revision" "Revision" LEFT JOIN "note" "Revision__Revision_note" ON "Revision__Revision_note"."id"="Revision"."noteId" WHERE ("Revision__Revision_note"."id" = 1)) "distinctAlias" ORDER BY "distinctAlias"."Revision_createdAt" DESC, "distinctAlias"."Revision_id" DESC, "Revision_id" ASC LIMIT 1
```

### Config

```shell
# Debug
NODE_ENV=development
HD_LOGLEVEL="debug"

# General settings
HD_BASE_URL="http://localhost:8080"
HD_SESSION_SECRET="this-should-be-a-long-and-secure-secret"

# Database settings
HD_DATABASE_TYPE="sqlite"
HD_DATABASE_NAME="./hedgedoc.sqlite"

# Uploads
HD_MEDIA_BACKEND="filesystem"
HD_MEDIA_BACKEND_FILESYSTEM_UPLOAD_PATH="uploads/"

# Auth
HD_AUTH_LOCAL_ENABLE_LOGIN="true"
HD_AUTH_LOCAL_ENABLE_REGISTER="true"
```

### Your Setup

- Host OS: Ubuntu 23.04
- NodeJS version: 20.3.0
- HedgeDoc version: f29a30d94b615c4e4f3bc9e5aedc4f47db75dd6d

**Comment by mrdrogdrog** [comment]
Does that happen EVERY time for you? Because I've observed it only for new notes.

**Comment by mrdrogdrog** [comment]
Fixed with #4271
